src/NewRecipePage.tsx

```diff
@@ -0,0 +1,495 @@
+              <Form.Item
+                name={[recipe.id]}
+                valuePropName="checked"
+                initialValue={false}
+                style={{
+                  margin: "0px",
+                  width: "100%",
+                  height: "100%",
+                  alignItems: "center",
+                  paddingTop: "10px",
+                  paddingBottom: "10px",
+                }}
+              >
+                <Checkbox
+                  style={{
+                    paddingLeft: "10px",
+                    width: "100%",
+                    height: "100%",
+                    alignItems: "center",
+                  }}
+                >
+                  {recipe.thumbnail.image && (
+                    <Image
+                      width={50}
+                      height={50}
+                      src={recipe.thumbnail.image}
+                    />
+                  )}
+                  <span style={{ marginLeft: "10px" }}>{recipe.name}</span>
+                </Checkbox>
+              </Form.Item>
+            </div>
+          ))}
+        </Form>
+      </Modal>
+    );
+  };
+
+  const deleteRecommendation = (recIndex: number) => {
+    const newLoadedRecipes: Recipe[] = [...loadedRecipes];
+    newLoadedRecipes.splice(recIndex, 1);
+    setLoadedRecipes(newLoadedRecipes);
+  };
+
+  const submitRecipe = async (values: any) => {
+    var image = "";
+    if (values.thumbnail !== undefined) {
+      image = values.thumbnail;
+    }
+
+    const recommendArray: string[] = [];
+    if (loadedRecipes.length > 0) {
+      loadedRecipes.map((recipe, index) =>
+        recommendArray.push(recipe.id.toString())
+      );
+    } else {
+      recommendArray.push("");
+    }
+
+    const ingredientArray: { starred: boolean; ingredient: string }[] = [];
+    for (const [key] of Object.entries(values.ingredients)) {
+      if (values.ingredients[key].starred === 1) {
+        ingredientArray.push({
+          starred: true,
+          ingredient: values.ingredients[key].ingredient,
+        });
+      } else {
+        ingredientArray.push({
+          starred: false,
+          ingredient: values.ingredients[key].ingredient,
+        });
+      }
+    }
+
+    const stepArray: { step: number; directions: string; image: string }[] = [];
+    for (const [key] of Object.entries(values.steps)) {
+      if (values.steps[key].image) {
+        stepArray.push({
+          step: parseInt(key) + 1,
+          image: values.steps[key].image,
+          directions: values.steps[key].directions,
+        });
+      } else {
+        stepArray.push({
+          step: parseInt(key) + 1,
+          image: "",
+          directions: values.steps[key].directions,
+        });
+      }
+    }
+
+    const current = new Date();
+    const date = `${
+      current.getMonth() + 1
+    }/${current.getDate()}/${current.getFullYear()}`;
+
+    const newRecipe: Recipe = {
+      id: allRecipes[allRecipes.length - 1].id + 1,
```

Comment:
This will cause a bug where if 2 people load the page at the same time and submit recipes (not even at the same time), they will end up with the same ID.  You should always set ID's on the server, where there's less chance of a race condition.

Didn't you say you wanted to switch to mongodb IDs anyway?